server/services/offender/index.js

```diff
@@ -247,6 +277,27 @@ const createOffenderService = (
     return Timetable.create({ startDate, endDate }).build();
   }
 
+  async function getPrisonDetailsFor(prisonId) {
+    try {
+      logger.info(
+        `OffenderService (getPrisonDetailsFor) - prisonId: ${prisonId}`,
+      );
+
+      if (!prisonId) {
+        throw new Error('No prisonId passed');
+      }
+
+      const response = await repository.getPrisonDetailsFor(prisonId);
+      return Prison.from(response).format();
+    } catch (e) {
+      logger.error(
+        `OffenderService (getPrisonDetailsFor) - Failed: ${e.message} - prisonId: ${prisonId}`,
+      );
+      logger.debug(e.stack);
+      return null;
+    }
+  }
```

Comment:
Mirroring the comment on `getTransactionsFor` (but GitHub won't let me link because it's a draft review)... What motivates the split between a repository and service here? 

Could we keep the response error-handling logic alongside the API call in the repository? Splitting the two means we're bleeding concerns out into surrounding functions and tightly coupling them.